lib/prism.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const Prism = require('prismjs');
+const prismLoadLanguages = require('prismjs/components/');
+
+// https://github.com/PrismJS/prism/issues/2145
+const prismComponents = require('prismjs/components');
+const prismSupportedLanguages = Object.keys(prismComponents.languages);
+
+const escapeHTML = require('./escape_html');
+const unescapeHTML = require('./unescape_html');
+
+/**
+ * Wrapper of Prism.highlight()
+ * @param {String} code
+ * @param {String} language
+ */
+function prismHighlight(code, language) {
+  // Prism has not load the language pattern
+  if (!Prism.languages[language] && prismSupportedLanguages.includes(language)) prismLoadLanguages(language);
+
+  if (Prism.languages[language]) {
+    // Prism escapes output by default
+    return Prism.highlight(unescapeHTML(code), Prism.languages[language], language);
+  }
+
+  // Current language is not supported by Prism, return origin code;
+  return escapeHTML(code);
+}
+
+/**
+ * Generate line number required HTML snippet
+ * @param {String} code - Highlighted code
+ */
+function lineNumberUtil(code) {
+  const matched = code.match(/\n(?!$)/g);
+  const num = matched ? matched.length + 1 : 1;
+  const lines = new Array(num + 1);
```

Comment:
I think this code will cause misreading.
The variable `lines` should be a sequence of `<span></span>`.
The code `new Array(num + 1)` is only a trick to use an edge case of `Array.prototype.join()`.
Please keep original codes as you can.
https://github.com/PrismJS/prism/blob/master/plugins/line-numbers/prism-line-numbers.js

Reply:
Updated and rebased.